XLS-0081d-permissioned-dex/README.md

```diff
@@ -0,0 +1,593 @@
+
+| Field Name | Always Present? | JSON Type | Description |
+|------------|-----------------|-----------|------------|
+|`alternatives`|✔️|`array`|An array of objects with suggested paths to take, as described below. If empty, then no paths were found connecting the source and destination accounts.|
+|`destination_account`|✔️|`string`|The address of the account that would receive a transaction.|
+|`destination_amount`|✔️|`string` or `object`|The currency amount that the destination would receive in a transaction.|
+|`source_account`|✔️|`string`|The address that would send a transaction.|
+|`full_reply`|✔️|`boolean`|If `false`, this is the result of an incomplete search. A later reply may have a better path. If `true`, then this is the best path found. (It is still theoretically possible that a better path could exist, but `rippled` won't find it.) Until you close the pathfinding request, `rippled` continues to send updates each time a new ledger closes.|
+
+This proposal puts forward the following addition:
+
+| Field Name | Required? | JSON Type | Description |
+|------------|-----------|-----------|-------------|
+|`domain`| |`string`|The object ID of a `PermissionedDomain` object, if the orderbook shown is for a specific domain.|
+
+## 8. RPC: `ripple_path_find`
+
+The [`ripple_path_find` RPC method](https://xrpl.org/ripple_path_find.html) already exists on the XRPL. This proposal suggests some modifications to also support permissioned DEX domains.
+
+### 8.1. Request Fields
+
+As a reference, here are the fields that `ripple_path_find` currently accepts:
+
+| Field Name | Required? | JSON Type | Description |
+|------------|-----------|-----------|-------------|
+|`source_account`|✔️|`string`|The address of the account that would send funds in a transaction|
+|`destination_account`|✔️|`string`|The address of the account that would receive funds in a transaction|
+|`destination_amount`|✔️|`string` or `object`|The currency amount that the destination account would receive in a transaction.|
+|`send_max`| |`string` or `object`|The currency amount that would be spent in the transaction. Cannot be used with `source_currencies`.|
+|`source_currencies`| |`array`|An array of currencies that the source account might want to spend. Each entry in the array should be a JSON object with a mandatory `currency` field and optional `issuer` field, like how currency amounts are specified.|
+|`ledger_hash`| |`string`|A 20-byte hex string for the ledger version to use. |
+|`ledger_index`| |`string` or `number`|The ledger index of the ledger to use, or a shortcut string to choose a ledger automatically.|
+
+This proposal puts forward the following addition:
+
+| Field Name | Required? | JSON Type | Description |
+|------------|-----------|-----------|-------------|
+|`domain`| |`string`|The object ID of a `PermissionedDomain` object. If this field is included, then the paths will be filtered to only show the valid paths for that domain.|
```

Comment:
this would potentially introduce fundamental changes to the OrderBookDB that fetches currency pair using hash map. At this point i'm not sure of the feasibility and effort to implement this change. Pathfinding would be more efficient if this is implemented. 

However another option(also to be confirmed if this is viable) is to leave OrderBookDB as it is, and then filter out paths during liquidity calculation using DomainID. This is less efficient because we will be calculating liquidity for paths that don't have a domainID.